jetty-util-ajax/src/main/java/org/eclipse/jetty/util/ajax/JSON.java

```diff
@@ -93,12 +92,23 @@
     public final static JSON DEFAULT = new JSON();
 
     private Map<String, Convertor> _convertors = new ConcurrentHashMap<String, Convertor>();
+    private StringEncoder _stringEncoder = new JSONDefaultStringEncoder();
```

Comment:
This should be final and initialized with constructor injection

Reply:
Can't do it effectively through the constructor, as JSON is essentially a singleton. :-(

Reply:
Then  a setter is even worse, as that can change encoding for other threads using the singleton.   Perhaps it needs to be an optional parameter to the methods on JSON?